src/plots.jl

```diff
@@ -1,3 +1,289 @@
+              tickfontsize = 10, 
+              guidefontsize = 15, 
+              title = plot_title,
+              legendfontsize = 15, 
+              legendtitlefont = 15,
+              titlefont = 16
+    )
+    
+    plt = Plots.plot(s, dNds.*s.*s, linewidth = 4; kwargs...)
+
+    if mark_max
+        dNds = dNds.*s.*s
+        p0 = findfirst(smax -> smax == maximum(dNds), dNds)
+        Plots.plot!(plt, [s[p0]], [dNds[p0]], seriestype = :scatter, 
+                                    markersize = 7, 
+                                    markercolor = :black,
+                                    label = false
+        )
+    end
+
+    return plt
+end
+
+# energy spectra s² dN/ds = f(s = log₁₀(E/MeV)) for particles with mass
+# at the moment, time has to be added by hand
+function energy_spectrum(sim; mass = m_e, species = "electron", plot_title = "Electron energy spectrum")
+    file = sim[end]
+    px, py, pz = file["px/$species", "py/$species", "pz/$species"]
+
+    #should be implemented: read species mass from input.deck
+    E₀ = sqrt.(mass^2*c_0^4 .+ (px.^2 + py.^2 + pz.^2) * c_0^2)
+    s₀ = log.(ustrip(u"MeV", E₀))
+
+    h = fit(Histogram, s₀; nbins = 200)
+    s = midpoints(collect(h.edges[1]))
+    N = h.weights
+    dNds = N./abs(s[1]-s[2]) 
+
+    idx = findall(x -> x > 0, s)
+    s = s[idx]
+    dNds = dNds[idx]
+
+    kwargs = (xlabel = "s = log₁₀(E/MeV)", 
+              ylabel = "s² dN/ds", 
+              tickfontsize = 10, 
+              guidefontsize = 15, 
+              title = plot_title,
+              legendfontsize = 15, 
+              legendtitlefont = 15,
+              titlefont = 16
+    )
+    
+    plt = Plots.plot(s, dNds.*s.*s, linewidth = 4; kwargs...)
+
+    return plt
+end
+
+# Angular distribution of gamma energy emission (MeV/srad)
+# angles are given by the direction of the momentum (the reference axis is Ox)
+# one can filter photons in cone. default angle is pi (180⁰), i.e. all photons going to the right
+# ! angle argument is given in radians
+function photon_solid_angle_emission(sim;  angle = pi, species = "photon", plot_title = "Solid angle emission")
+    file = sim[end]
+
+    px, py, pz= file["px/$species",
+                     "py/$species",
+                     "pz/$species"
+    ]
+
+    idx = findall(p -> p > 0, ustrip.(u"kg*m/s",px))
+    px = px[idx]
+    py = py[idx]
+    pz = pz[idx]
+    
+    pₚₑᵣₚ = sqrt.(py.^2 + pz.^2)   #transversal momentum of each photon
+    θ = atan.(pₚₑᵣₚ./px)           #emission angles for each photon
+    ϕ = atan.(py, pz)
+
+    # filtering data: keep photons that propagate within 2θ
+    idx = findall(θ₀ -> θ₀ < angle/2, θ)
+    px = px[idx]
+    py = py[idx]
+    pz = pz[idx]
+    θ = θ[idx]
+    ϕ = ϕ[idx]
+
+    Eᵧ = sqrt.(px.^2 + py.^2 + pz.^2) * c_0
+
+    data = (ϕ .+ π, θ)
+    w = weights(ustrip.(u"MeV", Eᵧ))
+
+    h = fit(Histogram, data, w, nbins = (360,90))
+    xedges = collect(h.edges[1]) 
+    yedges = collect(h.edges[2]) *180/pi
```

Comment:
You could also use `rad2deg`
```suggestion
    yedges = rad2deg.(h.edges[2])
```

Reply:
Done